Content.Server/Chemistry/EntitySystems/SolutionInjectWhileEmbeddedSystem.cs

```diff
@@ -0,0 +1,53 @@
+using Content.Server.Body.Components;
+using Content.Server.Body.Systems;
+using Content.Server.Chemistry.Components;
+using Content.Shared.Chemistry.EntitySystems;
+using Content.Shared.Chemistry.Events;
+using Content.Shared.Inventory;
+using Content.Shared.Popups;
+using Content.Shared.Projectiles;
+using Content.Shared.Tag;
+using Content.Shared.Weapons.Melee.Events;
+using Robust.Shared.Collections;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Chemistry.EntitySystems;
+
+/// <summary>
+/// System for handling the different inheritors of <see cref="BaseSolutionInjectOnEventComponent"/>.
+/// Subscribes to relevent events and performs solution injections when they are raised.
+/// </summary>
+public sealed class SolutionInjectWhileEmbeddedSystem : EntitySystem
+{
+	[Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly BloodstreamSystem _bloodstream = default!;
+    [Dependency] private readonly InventorySystem _inventory = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SharedSolutionContainerSystem _solutionContainer = default!;
+    [Dependency] private readonly TagSystem _tag = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+    }
+
+	public override void Update(float frameTime)
+	{
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<SolutionInjectWhileEmbeddedComponent, EmbeddableProjectileComponent>();
+        while (query.MoveNext(out var uid, out var injectComponent, out var projectileComponent))
+        {
+            if (_gameTiming.CurTime < injectComponent.NextUpdate)
+                continue;
+            if(projectileComponent.EmbeddedIntoUid == null) {
+                continue;
+            }
+
+            var ev = new InjectOverTimeEvent(projectileComponent.EmbeddedIntoUid.Value);
+			RaiseLocalEvent(uid, ref ev);
+
+            injectComponent.NextUpdate = _gameTiming.CurTime + injectComponent.UpdateInterval;
+		}
+	}
+}
```

Comment:
You should never do curtime + interval that causes rounding problems, just add on the update interval.

Additionally this should be set on MapInit, also these need to be cleaned up or they will be a performance sink for the rest of a round.

Reply:
Can you elaborate on what "Cleaned up" means?
As in do something on component shutdown?
If destroyed then I don't think they can be since there will be a limited amount of them provided anyways, would be ass to only have like 3 and lose them after use. (Unless it ends up being available to all crew and not only traitors)